### PR TITLE
added controller, dto, mapper

### PR DESCRIPTION
**Pull request by SvNadorozhnyi**

### CODE REVIEW

File: src/main/java/org/example/bookstore/dto/BookDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.example.bookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (okuzan) [inline src/main/java/org/example/bookstore/dto/BookDto.java]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -47,6 +49,16 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-data-jpa</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+            <version>3.3.5</version>
```
Comment (Elena-Bruyako) [inline pom.xml]: remove version
```suggestion
```

File: src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -52,4 +53,12 @@ public List<Book> findAll() {
             throw new RuntimeException("Can't get all books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java:64]: add catch block

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
just two comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**